### ISSUE TITLE
css / scss support

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/feedback#105]

**Comment by jamesdbruner** [comment]
I'd love some scss syntax highlighting, I'm assuming that's what's meant by support but wanted to clarify incase that wasn't clear.

**Comment by Atelier-Mirai** [comment]
We hope that CSS and SCSS will be supported.

**Comment by maxbrunsfeld** [comment]
CSS support will land in Zed v0.59. I'm just going to change the title of this issue to focus on SCSS, so folks who are waiting on SCSS can continue to subscribe here.

**Comment by guttenbergovitz** [comment]
I am using 0.8.x v and css seems to be available on the level of treesitter not langserver what is sort of show stopper.

**Comment by iamnbutler** [comment]
I implemented most of this, but killed the PR since the tree-sitter grammar is filled with errors. Either we or someone else will need to work through updating the grammar before we can ship this.

https://github.com/serenadeai/tree-sitter-scss

I only got about 1/8th of the way through testing the examples in the SCSS docs before it was obvious upstream changes to the grammar will be needed.

If anyone wants to pick that up here were my notes so far.

```
Fix:
- The sibling selector `~` causes an ERROR
- Can't interpolate property names with `-`/`-#{$prefix}` (ERROR)
- Custom property names aren't marked with `function_name` or similar to allow correct styling.

Nice to have:
- support doc comments (///) (should be `doc_comment`
```

Otherwise I'll pick this back up after a few of the ones with accurate grammars are finished.
